src/generate/constants.rs

```diff
@@ -4,6 +4,13 @@ use il2cpp_metadata_raw::Il2CppTypeDefinition;
 pub trait TypeExtentions {
     fn is_static(&self) -> bool;
     fn is_const(&self) -> bool;
+    fn is_public_method(&self) -> bool;
+    fn is_abstract_method(&self) -> bool;
+    fn is_static_method(&self) -> bool;
+    fn is_virtual_method(&self) -> bool;
+    fn is_param_in(&self) -> bool;
+    fn is_param_out(&self) -> bool;
+    fn is_param_optional(&self) -> bool;
```

Comment:
These are not attributes on the _type_ but rather the specific definition. Ex, the various method extensions should be checking against the `Il2CppMethodDefinition.flags` field

Reply:
And for in/out/byref, you just need to check the byref field